CrudApp/services/data/BookRepository.cs

```diff
@@ -0,0 +1,256 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dapper;
+using Microsoft.EntityFrameworkCore;
+using CrudApp.models;
+using Newtonsoft.Json;
+
+// when setting up database naming schemes, you can say all camelcase is converted to SQL conventions
+
+namespace CrudApp.services.data
+{
+    public interface IBookRepository
+    {
+        // create a test for reading database
+        // public unneccessary 
+        // the classes based on this interface only return from database, rather than return and print. so readdatabase shouldn't be void 
+        // the interface class should deal with the printing 
+        List<Book> ReadDatabase();
+        (Book? book, List<Book>? databaseBooks, string? message) GetBookByTitle(string titleOfBook);
+        (bool isSuccess, string? message) AddBook(Book book);
+        (bool isSuccess, string? message) UpdateBook(string titleOfBookToUpdate, Book updatedBook);
+        (bool isSuccess, string? message) RemoveBook(string titleOfBookToRemove);
+    }
+
+    public class LibraryContext : DbContext
+    {
+        public DbSet<Book> simple_library { get; set; }
+
+        public static string connectionString = "Server=localhost;User ID=sa;Password=9n8kZ81J0iuB;Initial Catalog=SIMPLE_LIBRARY;Integrated Security=false;TrustServerCertificate=True";
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer(connectionString);
+        }
+
+    }
+
+    public class FileDbConnection : IBookRepository
+    {
+        private IFileService _fileService;
+        private readonly string _filePath;
+
+        public FileDbConnection(IFileService fileService, string filePath)
```

Comment:
Yeah - we should run through this one together as more complicated than it needs to be